src/kcas.mli

```diff
@@ -45,6 +45,25 @@ module Loc : sig
   (** [decr r] atomically decrements [r]. *)
 end
 
+(** Operating modes of the [k-CAS-n-CMP] algorithm. *)
+module Mode : sig
+  type t
+  (** Type of an operating mode of the [k-CAS-n-CMP] algorithm. *)
+
+  val lock_free : t
+  (** In [lock_free] mode the algorithm makes sure that at least one domain will
+      be able to make progress. *)
+
+  val obstruction_free : t
+  (** In [obstruction_free] mode the algorithm proceeds optimistically and
+      allows operations to fail due to {!Interference} from other domains that
+      might have been prevented in the {!lock_free} mode. *)
```

Comment:
We should highlight that interference can look spurious, i.e. value of some loc does not have to change for interference to be raised.

Reply:
I added a commit with a bit of documentation on this.